src/python/tests/genome_metadata/test_dump.py

```diff
@@ -115,34 +119,129 @@ def test_check_genebuild_version(
+        ),
+        pytest.param(
+            {"genebuild": {"new_key": "_"}}, {"genebuild": {}}, None, False, id="Filters on '_' value"
+        ),
+        pytest.param({"BRC5": "new_value"}, {}, None, False, id="BRC5 new value"),
+        pytest.param(
+            {"meta": "key", "species": {"alias": "woof"}},
+            {"species": {"alias": "woof"}},
+            None,
+            False,
+            id="Test alias",
+        ),
+        pytest.param(
+            {"added_seq": {"region_name": [1, 2]}},
+            {"added_seq": {"region_name": ["1", "2"]}},
+            None,
+            False,
+            id="Added seq region_name",
+        ),
+        pytest.param({}, {}, None, False, id="BRC5 new value"),
+        pytest.param(
+            {
+                "species": {
+                    "display_name": "Honeybee",
+                    "annotation_source": "Ensembl",
+                    "production_name": "apis_melifera_gca123v1",
+                    "scientific_name": "apis_melifera",
+                    "taxonomy_id": "70921",
+                }
+            },
+            {
+                "species": {
+                    "display_name": "Honeybee",
+                    "production_name": "apis_melifera_gca123v1",
+                    "taxonomy_id": "70921",
+                }
+            },
+            "species_filter.json",
+            False,
+            id="Filter via input meta JSON",
+        ),
+        pytest.param(
+            {
+                "annotation": {
+                    "provider_name": "ENA",
+                },
+                "assembly": {"accession": "GCA_000111222.3", "version": "1"},
+                "genebuild": {"method": "import", "version": "1"},
+            },
+            {"assembly": {"version": "1"}, "genebuild": {"method": "import", "version": "1"}},
+            "version_filter.json",
+            False,
+            id="Asm + Genebuild version filter",
+        ),
+        pytest.param(
+            {
+                "annotation": {
+                    "provider_name": "ENA",
+                },
+                "assembly": {"accession": "GCA_000111222.3", "version": "1"},
+                "genebuild": {"method": "import", "version": "1"},
+            },
+            {"genebuild": {"method": "import"}},
+            "filter_noupdate.json",
+            True,
+            id="Only geneBuild method, restrict update",
+        ),
     ],
 )
-def test_filter_genome_meta(genome_metadata: Dict[str, Any], output: Dict[str, Any]) -> None:
+def test_filter_genome_meta(
+    data_dir: Path,
+    genome_metadata: Dict[str, Any],
+    output: Dict[str, Any],
+    meta_filter: StrPath,
+    meta_update: bool,
+) -> None:
     """Tests the `dump.filter_genome_meta()` method.
 
     Args:
         genome_metadata: Nested genome metadata key values.
         output: Expected change in the genome metadata dictionary.
+        meta_filter:
+        restrict_filter:
```

Comment:
Missing descriptions.
```suggestion
        filter_file:
        restrict_filter:
```